### PR TITLE
Hide discrete logarithm k such that H = kG

### PR DESCRIPTION
**Pull request by Acentelles**
Address issue https://github.com/adjoint-io/aos-signature/issues/5

### CODE REVIEW

File: src/LSAG.hs [hunk 1]
```diff
@@ -116,8 +117,10 @@ verify pubKeys (ch0, [s], y) msg = panic "Invalid input"
 verify pubKeys (ch0, s0:s1:s2ToEnd, y) msg = ch0 == ch0'
   where
     curve0 = ECDSA.public_curve $ head pubKeys
-    -- h = [H(L)] * g
-    h = ECC.pointBaseMul curve0 (hashPubKeys curve0 pubKeys)
+    -- h = [H(L)]
+    h = generateH g curve0 (show $ hashPubKeys curve0 pubKeys)
+
+    -- h = ECC.pointBaseMul curve0 (hashPubKeys curve0 pubKeys)
```
Comment (sdiehl) [inline src/LSAG.hs]: Probably don't need this anymore.

File: src/LSAG.hs [hunk 2]
```diff
@@ -204,6 +207,25 @@ pointToBS (ECC.Point x y) = show x <> show y
 pubKeysToBS :: [ECDSA.PublicKey] -> BS.ByteString
 pubKeysToBS = foldMap (pointToBS . ECDSA.public_q)
 
+
+-- | Iterative algorithm to generate H.
+-- The important to hide its discrete logarithm "k" such that H = kG
+generateH :: ECC.Point -> ECC.Curve -> [Char] -> ECC.Point
+generateH g curve extra =
+  case yM of
+    Nothing -> generateH g curve (toS $ '1':extra)
```
Comment (sdiehl) [inline src/LSAG.hs]: The use of this magic `'1'` character is a little odd. What's the context on this choice?

Comment (Acentelles) [reply src/LSAG.hs]: `x` is derived from the hash. The algorithm needs to iterate to generate a new `x` when there isn't a `y` s.t. `y = x^3 + 7 mod p`. We need to deterministically create a new hash from the existing hash. `1` is an arbitrary change to the existing hash

Comment (sdiehl) [reply src/LSAG.hs]: Ok, I see. This makes sense.

Comment (Acentelles) [reply src/LSAG.hs]: Do you have any suggestions to make it more sensible?

Comment (Acentelles) [reply src/LSAG.hs]: Added a name to the magic number

**Pull request merged**